packages/twenty-server/src/engine/modules/calendar/timeline-calendar-event.resolver.ts

```diff
@@ -0,0 +1,113 @@
+import { HttpException, HttpStatus, UseGuards } from '@nestjs/common';
+import {
+  Query,
+  Args,
+  ArgsType,
+  Field,
+  ID,
+  Int,
+  Resolver,
+} from '@nestjs/graphql';
+
+import { Max } from 'class-validator';
+
+import { User } from 'src/engine/modules/user/user.entity';
+import { AuthUser } from 'src/engine/decorators/auth/auth-user.decorator';
+import { AuthWorkspace } from 'src/engine/decorators/auth/auth-workspace.decorator';
+import { JwtAuthGuard } from 'src/engine/guards/jwt.auth.guard';
+import { TIMELINE_CALENDAR_EVENTS_MAX_PAGE_SIZE } from 'src/engine/modules/calendar/constants/calendar.constants';
+import { TimelineCalendarEventsWithTotal } from 'src/engine/modules/calendar/dtos/timeline-calendar-events-with-total.dto';
+import { TimelineCalendarEventService } from 'src/engine/modules/calendar/timeline-calendar-event.service';
+import { UserService } from 'src/engine/modules/user/services/user.service';
+import { Workspace } from 'src/engine/modules/workspace/workspace.entity';
+
+@ArgsType()
+class GetTimelineCalendarEventsFromPersonIdArgs {
+  @Field(() => ID)
+  personId: string;
+
+  @Field(() => Int)
+  page: number;
+
+  @Field(() => Int)
+  @Max(TIMELINE_CALENDAR_EVENTS_MAX_PAGE_SIZE)
+  pageSize: number;
+}
+
+@ArgsType()
+class GetTimelineCalendarEventsFromCompanyIdArgs {
+  @Field(() => ID)
+  companyId: string;
+
+  @Field(() => Int)
+  page: number;
+
+  @Field(() => Int)
+  @Max(TIMELINE_CALENDAR_EVENTS_MAX_PAGE_SIZE)
+  pageSize: number;
+}
+
+@UseGuards(JwtAuthGuard)
+@Resolver(() => TimelineCalendarEventsWithTotal)
+export class TimelineCalendarEventResolver {
+  constructor(
+    private readonly timelineCalendarEventService: TimelineCalendarEventService,
+    private readonly userService: UserService,
+  ) {}
+
+  @Query(() => TimelineCalendarEventsWithTotal)
+  async getTimelineCalendarEventsFromPersonId(
+    @AuthWorkspace() { id: workspaceId }: Workspace,
+    @AuthUser() user: User,
+    @Args()
+    { personId, page, pageSize }: GetTimelineCalendarEventsFromPersonIdArgs,
+  ) {
+    const workspaceMember = await this.userService.loadWorkspaceMember(user);
+
+    if (!workspaceMember) {
+      throw new HttpException(
```

Comment:
You can directly throw NotFoundException which is less verbose

Reply:
Actually my bad I forgot we were in context of graphql here! You should throw NotFoundError instead which is more aligned with gql (no 4xx status code)